lib/rubocop/cop/autocorrect_logic.rb

```diff
@@ -88,10 +90,28 @@ def surrounding_percent_array(offense_range)
         end
 
         percent_array.map(&:source_range).find do |range|
-          offense_range.begin_pos > range.begin_pos && range.overlaps?(offense_range)
+          range_overlaps_offense?(offense_range, range)
         end
       end
 
+      def string_continuation(offense_range)
+        return nil if offense_range.empty?
+
+        string_continuation_nodes = processed_source.ast.each_descendant.filter_map do |node|
+          range_by_lines(node.source_range) if string_continuation?(node)
+        end
+
+        string_continuation_nodes.find { |range| range_overlaps_offense?(offense_range, range) }
+      end
+
+      def range_overlaps_offense?(offense_range, range)
+        offense_range.begin_pos > range.begin_pos && range.overlaps?(offense_range)
```

Comment:
It seems the test passes even without `range.overlaps?(offense_range)`. Can you review the logic or the test?

Reply:
It is necessary. I added a second test that would fail without it:

```ruby
context 'and the offense is on a different line than a string continuation' do
  it 'adds a single one-line disable statement' do
    create_file('.rubocop.yml', <<~YAML)
      AllCops:
        DisabledByDefault: true
      Lint/DuplicateHashKey:
        Enabled: true
    YAML
    create_file('example.rb', <<~'RUBY')
      x = 'something'\
        'something else'
      {
        key1: 'foo',
        key1: 'bar'
      }
    RUBY
    expect(exit_code).to eq(0)
    expect(File.read('example.rb')).to eq(<<~'RUBY')
      x = 'something'\
        'something else'
      {
        key1: 'foo',
        key1: 'bar' # rubocop:todo Lint/DuplicateHashKey
      }
    RUBY
  end
end
```